From aadd16c5edfa06a0521b8e8f6343111a85189430 Mon Sep 17 00:00:00 2001
From: Baoan Du <baoan.du.xj@renesas.com>
Date: Sat, 8 Jul 2023 05:56:44 +0000
Subject: [PATCH 34/40] dts: renesas: Add device configuration for CR7 PoC

Support CR7 remote processor with RVGC functionality.
Remove native physical devices related to display pipelines that
are now handled by CR7.

The rest of devices (like sound, csi, VIN, etc.) are still
controlled by CA5X.

Signed-off-by: Sergii Piatakov <sergii.piatakov@globallogic.com>
Signed-off-by: Vladyslav Andrishko <vladyslav.andrishko@globallogic.com>
---
 .../renesas/r8a77965-salvator-xs-android.dts  |   3 +
 arch/arm64/boot/dts/renesas/r8a77965.dtsi     |   2 +-
 .../dts/renesas/salvator-xs_cr7_rproc.dtsi    | 113 ++++++++++++++++++
 3 files changed, 117 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/renesas/salvator-xs_cr7_rproc.dtsi

diff --git a/arch/arm64/boot/dts/renesas/r8a77965-salvator-xs-android.dts b/arch/arm64/boot/dts/renesas/r8a77965-salvator-xs-android.dts
index 792dd55c5680..9d1782f1b31d 100644
--- a/arch/arm64/boot/dts/renesas/r8a77965-salvator-xs-android.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77965-salvator-xs-android.dts
@@ -13,3 +13,6 @@
 #include "r8a77965-salvator-xs.dts"
 #include "android/salvator.dtsi"
 #include "android/r8a77965.dtsi"
+
+/************************ CR7 PoC Salvator XS M3N Specific ************************/
+#include "salvator-xs_cr7_rproc.dtsi"
diff --git a/arch/arm64/boot/dts/renesas/r8a77965.dtsi b/arch/arm64/boot/dts/renesas/r8a77965.dtsi
index 93f30624ca87..c7778163b702 100644
--- a/arch/arm64/boot/dts/renesas/r8a77965.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77965.dtsi
@@ -236,7 +236,7 @@
 		clock-frequency = <0>;
 	};
 
-	soc {
+	soc: soc {
 		compatible = "simple-bus";
 		interrupt-parent = <&gic>;
 		#address-cells = <2>;
diff --git a/arch/arm64/boot/dts/renesas/salvator-xs_cr7_rproc.dtsi b/arch/arm64/boot/dts/renesas/salvator-xs_cr7_rproc.dtsi
new file mode 100644
index 000000000000..316349e1c4d4
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/salvator-xs_cr7_rproc.dtsi
@@ -0,0 +1,113 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Device Tree Source for the Salvator-X 2nd version board
+ * Cortex-R7 (CR7) as remote processor specific extensions
+ *
+ * Copyright (C) 2020 Renesas Electronics Corp.
+ *
+ * This file is licensed under the terms of the GNU General Public License
+ * version 2.  This program is licensed "as is" without any warranty of any
+ * kind, whether express or implied.
+ */
+
+/ {
+	aliases {
+		/* The CR7 will control video input so exclusively uses i2c4
+		 * as it needs to access the adv7482, i.e. Linux can no longer
+		 * use i2c4 for anything */
+		/delete-property/ i2c4;
+	};
+
+	cpus {
+		/delete-property/ idle-states;
+	};
+
+	reserved-memory {
+		/delete-node/ rcar-ion;
+
+		/* Memory used by the CR7 */
+		cr7: linux,cr7@70000000 {
+			no-map;
+			reg = <0x0 0x70000000 0x0 0x10000000>;
+		};
+
+		ion_reserved: rcar-ion {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x00000000 0x80000000 0x0 0x18000000>; /* 384Mb */
+		};
+	};
+
+	cr7 {
+		compatible = "renesas,rcar-cr7";
+		memory-region = <&cr7>;
+	};
+
+	rvgc {
+		#address-cells = <2>;
+		#size-cells = <2>;
+
+		/* 1:1 mapping for the ion address space */
+		dma-ranges = <0x0 0x80000000 0x0 0x80000000 0x0 0x18000000>;
+
+		rvgc-memory {
+			memory-region = <&cma_reserved>;
+		};
+
+		displays {
+			display-0 {
+				display-map = <1>;
+				layers {
+					primary{
+						layer-map=<0x4>;
+						size-w=<1280>;
+						size-h=<720>;
+						pos-x=<320>;	// (1920 - 1280) / 2
+						pos-y=<180>;	// (1080 - 720) / 2
+						pos-override;
+					};
+				};
+			};
+		};
+	};
+
+	/delete-node/ cvbs-in;
+	/delete-node/ hdmi-in;
+	/delete-node/ hdmi0-out;
+	/delete-node/ lvds;
+	/delete-node/ vga;
+	/delete-node/ vga-encoder;
+};
+
+&soc {
+	mfis: mfis@e6260000 {
+		compatible = "renesas,mfis";
+		reg = <0x0 0xe6260000 0x0 0x200>;
+		interrupts = <0x00 0xe0 0x04 0x00 0xe1 0x04 0x00 0xe2 0x04 0x00 0xe3 0x04 0x00 0xe4 0x04 0x00 0xe5 0x04 0x00 0xe6 0x04 0x00 0xe7 0x04>;
+		interrupt-names = "eicr0", "eicr1", "eicr2", "eicr3", "eicr4", "eicr5", "eicr6", "eicr7";
+		renesas,mfis-channels = <0 1 2 3>;
+		status = "okay";
+
+		/delete-node/ mfis-as;
+	};
+};
+
+/delete-node/ &i2c4;
+/delete-node/ &fcpvd0;
+/delete-node/ &fcpvd1;
+/delete-node/ &vspd0;
+/delete-node/ &vspd1;
+/delete-node/ &du;
+/delete-node/ &lvds0;
+/delete-node/ &hdmi0;
+
+/delete-node/ &vin0;
+/delete-node/ &vin1;
+/delete-node/ &vin2;
+/delete-node/ &vin3;
+/delete-node/ &vin4;
+/delete-node/ &vin5;
+/delete-node/ &vin6;
+/delete-node/ &vin7;
+/delete-node/ &csi20;
+/delete-node/ &csi40;
-- 
2.25.1

